__tests__/app.test.js

```diff
@@ -233,5 +232,28 @@ describe("PATCH /api/articles/:article_id", () => {
         );
       });
   });
+});
+//DELETE /api/comments/:comment_id
+describe("DELETE /api/comments/:comment_id", () => {
+  test("should delete the comment and respond with status 204", () => {
+    return request(app).delete(`/api/comments/2`).expect(204);
+  });
 
+  test("should return 404 if the comment_id does not exist", () => {
+    return request(app)
+      .delete("/api/comments/345")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Comment not found");
+      });
+  });
+  test("should return 500 for internal server errors", () => {
+    jest.spyOn(db, "query").mockRejectedValueOnce(new Error("Database Error"));
+    return request(app)
+      .delete(`/api/comments/1`)
+      .expect(500)
+      .then((response) => {
+        expect(response.body.msg).toBe("Internal Server Error");
+      });
+  });
```

Comment:
This is not a valid test, we should never really expect to reach a 500 error code as it is a fallback error catcher, but all other error handlers should give us an error that is actually meaningful.